src/main/java/org/snomed/snowstorm/core/data/services/identifier/IdentifierCacheManager.java

```diff
@@ -154,6 +158,24 @@ public void populateIdBlock(IdentifierReservedBlock idBlock, int quantityRequire
 				cache.unlock();
 				requestSatisfied = true;
 			}
+		} else {
+			//If no cache available & not requesting for International (as already prefetched),
+			//then prefetch additional identifiers for subsequent requests.
+			int prefetchQuantity;
```

Comment:
When we create a cache it would be good to create for all component types in the namespace, as we do for International on startup

Reply:
@kaicode , I can update my code to include Axioms, but I'm not sure where you're seeing Axioms being included for International. In Config::getIdentifierCacheManager, only Concepts, Descriptions and Relationships are added to the cache. Shall I update that to include Axioms?

Reply:
There are no SCTIDs for axioms. Maybe I am not reading this correctly but it looks like this block only creates a cache for the component type being requested. I think if any type is requested we could create a cache and prefetch ids for all types straight away.

Reply:
@kaicode , ah I get you now. Great spot. Yes, you are correct in that the cache is only updated for one given component type. However, the `idBlock` variable is from `IdentifierService::getReservedBlock` which calls the method for all of the component types: 

```
cacheManager.populateIdBlock(idBlock, conceptIds, namespace, ...);
cacheManager.populateIdBlock(idBlock, descriptionIds, namespace, ...);
cacheManager.populateIdBlock(idBlock, relationshipIds, namespace, ...);
```

Reply:
Got it. This looks good then 😄